console/src/main/java/org/matrix/console/EventEmitter.java

```diff
@@ -0,0 +1,40 @@
+package org.matrix.console;
+
+
+import android.util.Log;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class EventEmitter<T> {
+    private static final String LOG_TAG = "EventEmitter";
+
+    private Set<Listener<T>> mCallbacks;
+
+    public EventEmitter() {
+        mCallbacks = new HashSet<>();
+    }
+
+    public void register(Listener<T> cb) {
+        mCallbacks.add(cb);
+    }
+
+    public void unregister(Listener<T> cb) {
+        mCallbacks.remove(cb);
+    }
+
+    public void fire(T t) {
```

Comment:
Some docs here to state that the callbacks will be executed *synchronously* and *on the calling thread* would be good. You can do android-magic to call events on the UI thread. Off the top of my head, it's something like:
```java
Looper looper = Looper.getMainLooper();
Handler uiHandler = new Handler(looper);
uiHandler.post(new Runnable() {
  @Override
  public void run() {
    // stuff
  }
);
```
It may be worth doing this so we can at least be consistent on the behaviour regardless of which thread is calling `fire()`.